src/main/ruby/truffleruby/core/truffle/regexp_operations.rb

```diff
@@ -58,6 +58,14 @@ def self.match_from(re, str, pos)
       USE_TRUFFLE_REGEX = Truffle::Boot.get_option('use-truffle-regex')
     end
 
+    if Truffle::Boot.get_option('regexp-instrument-creation') || Truffle::Boot.get_option('regexp-instrument-match')
```

Comment:
get_option should be under `delay`, because otherwise the value is snapshotted at pre-init time.
I'll do it.

Reply:
I thought snapshots got invalidated on options changing?

Reply:
For language options, yes, but one of these is a context option and so it can be changed while still using the pre-initialized context.

Reply:
There is actually an automated check for this, but we need to get `jt build native` to work back in GHA first.